src/bridge/cockpitpeer.c

```diff
@@ -279,16 +287,20 @@ on_other_control (CockpitTransport *transport,
         g_source_remove (self->authorize_values_timeout);
       self->authorize_values_timeout = g_timeout_add (2*60*1000, cockpit_peer_delete_authorize_values, self);
 
-      startup_done (self, problem);
-
       if (problem)
         {
+          gchar *stderr = startup_take_stderr (self);
+          startup_done (self, problem, stderr);
+          g_free (stderr);
           cockpit_transport_close (transport, problem);
         }
       else
         {
           g_debug ("%s: received init message from peer bridge", self->name);
           self->inited = TRUE;
+          CockpitPipe *pipe = cockpit_pipe_transport_get_pipe (COCKPIT_PIPE_TRANSPORT (transport));
```

Comment:
It seems like we might drop some stderr output this way.  Shouldn't we be taking the existing buffer before calling `stop()` and forwarding it?  We could build that functionality into `stop()` itself in fact, since you already add code in cockpitpipe for forwarding newly-arrived data *after* stop() is called, so you could then share....

Reply:
Yes, good point.

Reply:
I have changed it so that we keep reading into the buffer (which ensures proper maintaining of the pipe's state). If a flag is set, the buffer is just dumped out to stderr again.